elyra/pipeline/kfp/kfp_authentication.py

```diff
@@ -0,0 +1,739 @@
+                                      f'\'{runtime_config_name}\' and try again.',
+                                      provider=self._type)
+        return None
+
+
+class DEXStaticPasswordAuthenticator(AbstractAuthenticator):
+    """
+    Authenticator for DEX/static passwords
+    """
+
+    _type = SupportedAuthProviders.DEX_STATIC_PASSWORDS
+
+    def authenticate(self,
+                     kf_endpoint: str,
+                     runtime_config_name: str,
+                     username: str,
+                     password: str) -> Optional[str]:
+        """
+        Authenticate using static password authentication. An AuthenticationError is raised
+        if (1) kf_endpoint is unsecured (2) kf_endpoint does not
+        support static password authentication (3) the credentials are invalid
+
+        :param kf_endpoint: Kubeflow API endpoint to verify
+        :type kf_endpoint: str
+        :param runtime_config_name: Runtime configuration name where kf_endpoint is specified
+        :type runtime_config_name: str
+        :param username: Id to be used for authentication
+        :type username: str
+        :param password: Password to be used for authentication
+        :type password: str
+        :raises AuthenticationError: Authentication failed due to the specified error.
+        :return: A cookie value
+        """
+
+        # This code can be removed after the kfp runtime schema enforces that the values
+        # for username and password are valid
+        if _empty_or_whitespaces_only(username) or\
+           _empty_or_whitespaces_only(password):
+            raise AuthenticationError(f'Credentials are required to perform this type of authentication. '
+                                      f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                      provider=self._type)
+
+        with requests.Session() as s:
+
+            request_history = []
+
+            ################
+            # Determine if Endpoint is Secured
+            ################
+            resp = s.get(kf_endpoint, allow_redirects=True)
+            request_history.append((kf_endpoint, resp))
+            if resp.status_code != HTTPStatus.OK:
+                raise AuthenticationError(f'Error detecting whether Kubeflow server at {kf_endpoint} is secured: '
+                                          f'HTTP status code {resp.status_code}'
+                                          f'Update runtime configuration \'{runtime_config_name}\' and try again.',
+                                          provider=self._type,
+                                          request_history=request_history)
+
+            if len(resp.history) == 0:
+                # if we were NOT redirected, then the endpoint is UNSECURED
+                # treat this as an error.
+                raise AuthenticationError(f'The Kubeflow server at {kf_endpoint} is not secured using DEX with LDAP. '
```

Comment:
This is the `DEXStaticPasswordAuthenticator(`, you probably did not mean to say "DEX with LDAP"